src/core/src/op/reverse.cpp

```diff
@@ -53,107 +69,70 @@ void ov::op::v1::Reverse::validate_and_infer_types() {
     set_output_type(0, get_input_element_type(0), output_shape);
 }
 
-std::shared_ptr<ov::Node> ov::op::v1::Reverse::clone_with_new_inputs(const OutputVector& new_args) const {
+std::shared_ptr<ov::Node> Reverse::clone_with_new_inputs(const OutputVector& new_args) const {
     OV_OP_SCOPE(v1_Reverse_clone_with_new_inputs);
     check_new_args_count(this, new_args);
-    return std::make_shared<op::v1::Reverse>(new_args.at(0), new_args.at(1), m_mode);
+    return std::make_shared<Reverse>(new_args.at(0), new_args.at(1), m_mode);
 }
 
-ov::op::v1::Reverse::Mode ov::op::v1::Reverse::mode_from_string(const std::string& mode) const {
+Reverse::Mode Reverse::mode_from_string(const std::string& mode) const {
     static const std::map<std::string, Mode> allowed_values = {{"index", Mode::INDEX}, {"mask", Mode::MASK}};
 
     NODE_VALIDATION_CHECK(this, allowed_values.count(mode) > 0, "Invalid 'mode' value passed in.");
 
     return allowed_values.at(mode);
 }
 
-OPENVINO_SUPPRESS_DEPRECATED_START
-namespace reverseop {
-template <ov::element::Type_t ET>
-void get_axes(ov::AxisSet& axes, const ngraph::HostTensorPtr& in) {
-    auto axes_indices = in->get_data_ptr<ET>();
-    size_t axes_rank = in->get_element_count();
-    std::copy(axes_indices, axes_indices + axes_rank, std::inserter(axes, axes.end()));
-}
-}  // namespace reverseop
-
-#define GET_AXES(a, ...)                                      \
-    case element::Type_t::a: {                                \
-        OV_OP_SCOPE(OV_PP_CAT3(get_reverse_axes, _, a));      \
-        reverseop::get_axes<element::Type_t::a>(__VA_ARGS__); \
-    } break;
-
-bool ov::op::v1::Reverse::evaluate_reverse(const HostTensorVector& outputs, const HostTensorVector& inputs) const {
-    AxisSet axes{};
-    if (get_mode() == op::v1::Reverse::Mode::INDEX) {
-        switch (inputs[1]->get_element_type()) {
-            GET_AXES(i8, axes, inputs[1]);
```

Comment:
@praasz please note that `GET_AXES` enables CC, so same behavior should be restored

Reply:
Yes and changed it intentionally.
Instead have custom implementation the existing one can be re-used, which will exist regarding CC features.
In standard build it will result in smaller bin-size as code exists already. In CC build the code will exist also.